torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -184,7 +185,15 @@ void PropagateShapeOnNode(Node * node) {
 
 void PropagateShapeOnBlock(Block * block) {
   for (Node * node : block->nodes()) {
-    PropagateShapeOnNode(node);
+    try {
+      PropagateShapeOnNode(node);
+    } catch(std::exception & e) {
+      if(auto sl = node->getSourceLocation()) {
+        sl->wrapAndRethrowException(e, "operation failed shape propagation");
```

Comment:
Maybe say something more like "shape mismatch". This will be seen by the end users, and they don't know about the shape propagation pass.

Reply:
Yeah, I want to fix this because this error will suck no matter what. Technically it can even fire when the original code would have never executed the line (e.g. if the line is guarded). What I want to do in the future is to just propagate DynamicType when it fails and let all of our failures happen in the interpreter.

Reply:
I don't want to do this now, though, because it is too easy to just have a mistake in the shape_propagation get covered up by catching std::exception. If ATen itself threw a particular exception for sizing errors, then we would be able to catch just that exception type, and I'd be more confident about hiding the error.